### PR TITLE
sql: add some compatibility notes for time related operations

### PR DESCRIPTION
**Pull request by breeswish**
@winkyao PTAL

### CODE REVIEW

File: sql/mysql-compatibility.md [hunk 1]
```diff
@@ -165,3 +165,22 @@ TiDB 支持 MySQL 5.7 中 **绝大多数的 SQL 模式**，以下几种模式除
...
+
+### 日期时间处理的区别
+
+#### 时区
+
+MySQL 中时区默认为本地时区，依赖于系统内置的当前的时区的规则（例如什么时候开始夏令时等）进行计算；在未[导入时区表数据](https://dev.mysql.com/doc/refman/8.0/en/time-zone-support.html#time-zone-installation)的情况下不能通过时区名称的方式指定时区。
+
+TiDB 不需要导入时区表数据也能使用所有时区名称，采用系统当前安装的所有时区规则计算（一般为 `tzdata` 包），且无法通过导入时区表数据的形式修改计算规则。
+
+注意：对于能下推到 TiKV 的时间相关表达式会由 TiKV 进行计算，TiKV 总是采用 TiKV 内置时区规则计算，而不依赖于系统所安装的时区规则。若系统安装的时区规则与 TiKV 内置的时区规则版本不匹配，则在少数情况下可能发生能插入的时间数据无法再读出来的问题。例如，若系统上安装了 tzdata 2018a 时区规则，则在时区设置为 Asia/Shanghai 或时区设置为本地时区且本地时区为 Asia/Shanghai 情况下，时间 `1988-04-17 02:00:00` 可以被正常插入 TiDB 3.0 RC.1，但该记录对于特定类型 SQL 则无法再读出来，原因是 TiKV 3.0 RC.1 依据的 tzdata 2018i 规则中该时间在 Asia/Shanghai 时区中不存在（夏令时时间后移一小时）。
```
Comment (yikeke) [inline sql/mysql-compatibility.md]:
```suggestion
> **注意：**
>
> 能下推到 TiKV 的时间相关表达式会由 TiKV 进行计算。TiKV 总是采用 TiKV 内置时区规则计算，而不依赖于系统所安装的时区规则。若系统安装的时区规则与 TiKV 内置的时区规则版本不匹配，则在少数情况下可能发生能插入的时间数据无法再读出来的问题。例如，若系统上安装了 tzdata 2018a 时区规则，则在时区设置为 Asia/Shanghai 或时区设置为本地时区且本地时区为 Asia/Shanghai 的情况下，时间 `1988-04-17 02:00:00` 可以被正常插入 TiDB 3.0 RC.1，但该记录对于特定类型 SQL 则无法再读出来，原因是 TiKV 3.0 RC.1 依据的 tzdata 2018i 规则中该时间在 Asia/Shanghai 时区中不存在（夏令时时间后移一小时）。
```

File: sql/mysql-compatibility.md [hunk 2]
```diff
@@ -165,3 +165,22 @@ TiDB 支持 MySQL 5.7 中 **绝大多数的 SQL 模式**，以下几种模式除
     + MySQL 中默认设置：
         + MySQL 5.7：`OFF`
         + MySQL 8.0：`ON`
+
+### 日期时间处理的区别
+
+#### 时区
+
+MySQL 中时区默认为本地时区，依赖于系统内置的当前的时区的规则（例如什么时候开始夏令时等）进行计算；在未[导入时区表数据](https://dev.mysql.com/doc/refman/8.0/en/time-zone-support.html#time-zone-installation)的情况下不能通过时区名称的方式指定时区。
```
Comment (yikeke) [inline sql/mysql-compatibility.md]:
```suggestion
MySQL 默认使用本地时区，依赖于系统内置的当前的时区规则（例如什么时候开始夏令时等）进行计算；且在未[导入时区表数据](https://dev.mysql.com/doc/refman/8.0/en/time-zone-support.html#time-zone-installation)的情况下不能通过时区名称来指定时区。
```
In this sentence, "MySQL" is the subject, not "The timezone of MySQL" (MySQL 中时区).

File: sql/mysql-compatibility.md [hunk 3]
```diff
@@ -165,3 +165,22 @@ TiDB 支持 MySQL 5.7 中 **绝大多数的 SQL 模式**，以下几种模式除
...
+
+TiDB 不需要导入时区表数据也能使用所有时区名称，采用系统当前安装的所有时区规则计算（一般为 `tzdata` 包），且无法通过导入时区表数据的形式修改计算规则。
```
Comment (yikeke) [inline sql/mysql-compatibility.md]:
```suggestion
TiDB 不需要导入时区表数据也能使用所有时区名称，采用系统当前安装的所有时区规则进行计算（一般为 `tzdata` 包），且无法通过导入时区表数据的形式修改计算规则。
```

**Comment by breeswish** [comment]
@yikeke PTAL, thanks!

**Pull request merged**